### DISCUSSION TITLE
MPP Pen writes when hovering

### DISCUSSION THREAD

**Discussion by salehjg** [xournalpp/xournalpp#5829] [General] [answered]
I have a Pavilion X360 14'' with HP pen (MPP).

Basically, before the pen touches the screen, it starts writing. This is not specific to XournalPP but happens in other programs as well.

I don't know if there is any *.conf file to set the desired thresholds but I was thinking this could be implemented in XournalPP by modifying how the minimum pressure constraint is applied.

Before I start my own fork of XournalPP, do you guys have any advice? Maybe a more elegant way to solve the issue?

Right now, the pen starts writing a good 4mm away from the screen which makes things practically unusable!

**Comment by GithubAnon0000** [comment]
> Basically, before the pen touches the screen, it starts writing. This is not specific to XournalPP but happens in other programs as well.

Not an answer but wanted to mention that my laptop (HP envy convertible) reacts to my stylus pen from approximatly 2 mm away. It only changes the cursor though and only draws once I actually touch the touchscreen.
I honestly think this shouldn't be handeled by applications though. Maybe some drivers / firmware or whatever else would need to be changed.

> I don't know if there is any *.conf file to set the desired thresholds

Maybe `preferences > Stylus > Pressure Sensitivity` is what you are looking for?

**Reply by salehjg** [reply]
Thank you for your kind reply.

I do think the same way. It should be fixed by HP. Unfortunately, their support is beyond unbelievable. I am trying to return the device but nobody's answering the lines!

Anyways, I tried my idea with the pressure thing and it is working well. 
https://github.com/xournalpp/xournalpp/compare/master...salehjg:xournalpp:feature-pressure

Here is a video of the problem: https://youtu.be/871tfYjZaH4?t=86

But this is not the solution, it is just a bandaid on a much bigger problem.

**Reply by salehjg** [reply, marked as answer]
Just wanted to mention to future readers that I found the problem to be the brand-new HP Pen MPP 2.0 . It was defective out-of-the-box. I purchased another pen (Lenovo Precision Pen 2.0) and it worked perfectly as it should.

The HP support is still asking for the OS version and Windows Touch Calibration which is infuriating honestly. Just avoid buying this pen if you don't want trouble.

**Reply by CarCoderPro** [reply]
> Thank you for your kind reply.
> 
> I do think the same way. It should be fixed by HP. Unfortunately, their support is beyond unbelievable. I am trying to return the device but nobody's answering the lines!
> 
> Anyways, I tried my idea with the pressure thing and it is working well. [master...salehjg:xournalpp:feature-pressure](https://github.com/xournalpp/xournalpp/compare/master...salehjg:xournalpp:feature-pressure)
> 
> Here is a video of the problem: https://youtu.be/871tfYjZaH4?t=86
> 
> But this is not the solution, it is just a bandaid on a much bigger problem.

Hi..I do face the same issue with my pavilion and mpp 2.0 tilt pen hovering issue.. [master...salehjg:xournalpp:feature-pressure](https://github.com/xournalpp/xournalpp/compare/master...salehjg:xournalpp:feature-pressure) what is this and what should i do to use that?

**Reply by GithubAnon0000** [reply]
>  master...salehjg:xournalpp:feature-pressure what is this and what should i do to use that?

That is a code change that salehjg applied to his fork to improve the behavior of the (faulty) MPP pen. If you want to try it, then here are the steps:
1. Fork [xournalpp](https://github.com/xournalpp/xournalpp).
2. Find the file `src/core/gui/inputdevices/PenInputHandler.cpp` and apply the change as done by salehjg in https://github.com/xournalpp/xournalpp/compare/master...salehjg:xournalpp:feature-pressure.
3. Build xournalpp.
4. Test if it's better.
5. If it improved things, maybe open a suggestion in xournalpp's issue tracker / discussions, too.

**Reply by CarCoderPro** [reply]
> > master...salehjg:xournalpp:feature-pressure what is this and what should i do to use that?
> 
> That is a code change that salehjg applied to his fork to improve the behavior of the (faulty) MPP pen. If you want to try it, then here are the steps:
> 
> 1. Fork [xournalpp](https://github.com/xournalpp/xournalpp).
> 2. Find the file `src/core/gui/inputdevices/PenInputHandler.cpp` and apply the change as done by salehjg in [master...salehjg:xournalpp:feature-pressure](https://github.com/xournalpp/xournalpp/compare/master...salehjg:xournalpp:feature-pressure).
> 3. Build xournalpp.
> 4. Test if it's better.
> 5. If it improved things, maybe open a suggestion in xournalpp's issue tracker / discussions, too.

should i change those values done by salehjg or can i download the file from the xournal website and use it directly without any changes?

**Reply by GithubAnon0000** [reply]
If you want to try the fix salehjg implemented, you'll have to change the code as salehjg did it.

**Reply by CarCoderPro** [reply]
ok thanks ill try since this is my 1st github contribution towards an open source  software.. could u pls help me a bit. i've downloaded the source code.zip and after changing the code how to check whether my pblm has been resolved and how to install the win app from tat source code file

**Reply by GithubAnon0000** [reply]
You have to fork it, then build it. I use linux so cannot help you with windows specific problems (and it has been a long while since I build xournalpp myself.). But everything you need should be documented here: https://github.com/xournalpp/xournalpp/blob/master/CONTRIBUTING.md#creating-a-fork.

**Reply by CarCoderPro** [reply]
ok then.. thank u so much :)

**Reply by GithubAnon0000** [reply]
Sure, you're welcome!

If you need additional help, you can probably open a discussion here and ask for help. At least in the past the people here were quite helpful (and they probably still are).
